app/lib/dartdoc/dartdoc_page.dart

```diff
@@ -0,0 +1,419 @@
+          addLinkRel: (href) {
+            final u = Uri.tryParse(href);
+            if (u != null && !u.shouldIndicateUgc) {
+              // TODO: Determine if there is a better way to do this.
+              //       It's probably reasonable that internal links don't
+              //       required ugc + nofollow.
+              return [];
+            }
+            // Add ugc and nofollow if the host isn't one of ours.
+            return ['ugc', 'nofollow'];
+          },
+        );
+
+    // HACK: Adding markdown-body style
+    final body = document.body;
+    if (body != null) {
+      // TODO: Modify our CSS to accomodate the styles output from dartdoc
+      body.querySelectorAll('.markdown').forEach((elem) {
+        if (!elem.classes.contains('markdown-body')) {
+          elem.classes.add('markdown-body');
+        }
+      });
+    }
+
+    // Extract breadcrumbs, because we need to tweak these on pub.dev
+    final breadcrumbs = document
+            .querySelector('header#title')
+            ?.querySelector('ol.breadcrumbs')
+            ?.children
+            .map((crumb) {
+          var href = crumb.querySelector('a')?.attributes['href'];
+          if (href != null && !_validLink(href)) {
+            href = null;
+          }
+          if (href != null) {
+            final u = Uri.tryParse(href);
+            if (u == null || u.shouldIndicateUgc) {
+              href = null; // breadcrumbs should never link externally!
+            }
+          }
+          return Breadcrumb(
+            title: crumb.text,
+            href: href,
+          );
+        }).toList() ??
+        <Breadcrumb>[];
+
+    final L = document.querySelector('#dartdoc-sidebar-left');
+    L?.querySelector('header')?.remove();
+    L?.querySelector('.breadcrumbs')?.remove();
+    final left = L?.innerHtml;
+
+    final R = document.querySelector('#dartdoc-sidebar-right');
+    final right = R?.innerHtml;
+
+    final C = document.querySelector('#dartdoc-main-content');
+    // HACK: Replace <section> with <div> to make sanitizeHtml happy
+    for (final section in C?.querySelectorAll('section') ?? <Element>[]) {
+      final div = Element.tag('div');
+      div.attributes = section.attributes;
+      section.reparentChildren(div);
+      section.replaceWith(div);
+    }
+    final content = C?.innerHtml;
+
+    return DartDocPage(
+      title: document.querySelector('head > title')?.text ?? '',
+      description: document
+              .querySelector('head > meta[name=description]')
+              ?.attributes['content'] ??
+          '',
+      breadcrumbs: breadcrumbs,
+      left: sanitize(left),
+      right: sanitize(right),
+      content: sanitize(content),
+    );
+  }
+
+  d.Node get _left => d.unsafeRawHtml(left);
+  d.Node get _right => d.unsafeRawHtml(right);
+  d.Node get _content => d.unsafeRawHtml(content);
+
+  d.Node _renderHead(DartDocPageOptions options) =>
+      d.element('head', children: [
```

Comment:
Should we merge this with the pub-dev `head` rendering?

Reply:
They look a bit different, I'm not sure we're quite ready for it. I realize there is a little duplication, but that's not new -- all the duplicated stuff was also duplicated in the old customization logic.